layout/layout.go

```diff
@@ -492,82 +492,24 @@ func findLayerWithSha(layers []v1.Layer, diffID string) (v1.Layer, error) {
 	return nil, fmt.Errorf("previous image did not have layer with diff id %q", diffID)
 }
 
-// mutateConfig mutates the provided v1.Image to have the provided v1.Config and wraps the result
-// into a layout.Image (requires for override methods like Layers()
+// mutateConfig mutates the provided v1.Image to have the provided v1.Config,
+// wraps the result into a layout.Image,
+// and sets it as the underlying image for the receiving layout.Image (required for overriding methods like Layers())
 func (i *Image) mutateConfig(base v1.Image, config v1.Config) error {
 	image, err := mutate.Config(base, config)
 	if err != nil {
 		return err
 	}
-	return i.mutateImage(image)
+	return i.setUnderlyingImage(image)
 }
 
-// mutateConfigFile mutates the provided v1.Image to have the provided v1.ConfigFile and wraps the result
-// into a layout.Image (requires for override methods like Layers()
+// mutateConfigFile mutates the provided v1.Image to have the provided v1.ConfigFile,
+// wraps the result into a layout.Image,
+// and sets it as the underlying image for the receiving layout.Image (required for overriding methods like Layers())
 func (i *Image) mutateConfigFile(base v1.Image, configFile *v1.ConfigFile) error {
 	image, err := mutate.ConfigFile(base, configFile)
 	if err != nil {
 		return err
 	}
-	return i.mutateImage(image)
-}
-
-// mutateImage wraps the provided v1.Image into a layout.Image
-func (i *Image) mutateImage(base v1.Image) error {
-	manifest, err := base.Manifest()
-	if err != nil {
-		return err
-	}
-	if validMediaTypes(manifest) {
-		i.Image = &Image{
-			Image: base,
-		}
-	} else {
-		// images has docker media types, we need to override them
-		newBaseImage, err := overrideMediaTypes(base)
-		if err != nil {
-			return err
-		}
-		i.Image = &Image{
-			Image: newBaseImage,
-		}
-	}
-	return nil
-}
```

Comment:
Moved to layout/new.go and renamed `setUnderlyingImage`. We may want to make a separate function that checks and updates media types, as we don't really need to do that on the level of the image after it is first instantiated.